### PR TITLE
Add if/else for `Parser::ValueArray`

### PR DESCRIPTION
**Pull request by no-reply**
For initial review:

```ruby
value_array.if { |arry| arry.field(:a_field) }
  .else { |arry|  arry.field(:alternate_field) }

value_array.if.field(:a_field).reject { |v| v == 'SKIP ME' }
  .else { |arry|  arry.field(:alternate_field) }
```

This addresses the use case in: https://github.com/dpla/heidrun-mappings/commit/5768026c078ab1a037d7eb043f4abd9b72a9f289#commitcomment-13144153

### CODE REVIEW

**Comment by no-reply** [comment]
The failures are due to dependency issues. Taking a look now.

File: lib/krikri/parser.rb [hunk 1]
```diff
@@ -212,7 +215,55 @@ def fields(*args)
...
           field(*Array(f))
         end
-        self.class.new(results.flatten)
+        self.class.new(results.flatten, @top)
+      end
+
+      ##
+      # Sets the top of the call chain to self and returns or yields self
+      #
+      # @example with method chain syntax
```
Comment (anarchivist) [inline lib/krikri/parser.rb]: :tada: great docs :memo:

**Comment by anarchivist** [comment]
Looks good. I think this tests should pass if you rebase on `develop`.

**Comment by anarchivist** [comment]
Meanwhile, assigning to @markbreedlove for a second look.

**Comment by markbreedlove** [comment]
Looks good to me.  The only thing I can think of at the moment is that it might be good to document the significance of `@top` in `ValueArray#initialize` or maybe the class itself, because of its importance.

**Comment by no-reply** [comment]
If the new docs look good, and we aren't concerned about handling for extra complex cases, this is ready.

Are we worried about testing, e.g.:

```ruby
some_node.field('a', 'b', 'c').else { |arry| arry.field('d', 'e') }.else { |arry| arry.field('f', 'g') }
some_node.if.field('a', 'b', 'c').if.reject { |v| is_bad(v) }.else { |arry| arry.field('d', 'e') }
```

And other such aberrations people may try?

**Comment by markbreedlove** [comment]
I don't think extra tests for those would be necessary.  This should be fine to squash.

**Comment by no-reply** [comment]
I'll add some further notes to the commit message, if you want to hold off a moment.

**Comment by markbreedlove** [comment]
ok

**Comment by no-reply** [comment]
done.

**Pull request merged**